umh-core/pkg/service/redpanda/redpanda.go

```diff
@@ -956,3 +962,133 @@ func formatMemory(memory int) string {
 
 	return fmt.Sprintf("%d%s", memory, units[unitIndex])
 }
+
+// updateRedpandaClusterConfig updates Redpanda's cluster config via HTTP PUT request
+func (s *RedpandaService) UpdateRedpandaClusterConfig(ctx context.Context, redpandaName string, configUpdates map[string]interface{}) error {
+	if s.httpClient == nil {
+		return fmt.Errorf("http client not initialized")
+	}
+
+	// If the context is done, return an error
+	if ctx.Err() != nil {
+		return ctx.Err()
+	}
+
+	// Construct the request body
+	requestBody := map[string]interface{}{
+		"upsert": configUpdates,
+		"remove": []string{},
+	}
+
+	// Convert to JSON
+	jsonBody, err := json.Marshal(requestBody)
+	if err != nil {
+		return fmt.Errorf("failed to marshal request body: %w", err)
+	}
+
+	// Construct the URL
+	url := "http://localhost:9644/v1/cluster_config"
```

Comment:
maybe it makes sense to create this somehow dynamically or use a const for this? or use the same generation how we fetch the config, to show that this is the same API call